lib/on_form/collection_wrapper.rb

```diff
@@ -51,12 +54,12 @@ def parse_collection_attributes(params)
         destroy = self.class.boolean_type.cast(attributes['_destroy']) || self.class.boolean_type.cast(attributes[:_destroy])
         if id = attributes['id'] || attributes[:id]
           if destroy
-            records_to_destroy << id.to_i if allow_destroy
+            records_to_destroy << id.to_i if options[:allow_destroy]
           else
-            records_to_update[id.to_i] = attributes.except('id', :id, '_destroy', :destroy) if allow_update
+            records_to_update[id.to_i] = attributes.except('id', :id, '_destroy', :destroy) if options[:allow_update] && !call_reject_if(attributes)
```

Comment:
could just have this in an if block as it's pretty long. Makes it a bit easier to read too.